File: UI/src/app/models/addCourseCommand.ts
import {Guid} from "../utils/CreateGuid";

export class AddCourseCommand {
  courseGuid: string;
  name: string;
  teacher: string;
  maxParticipants: number;
  constructor(name: string, maxParticipants: number, teacher: string) {
    this.courseGuid = Guid.newGuid();
    this.name = name;
    this.maxParticipants = maxParticipants;
    this.teacher = teacher;
  }
}

File: Dtos/Commands/AddCourseCommand.cs
namespace OnlineCourse.Api.Dtos.Commands
{
	using System;

	/// <summary>
	/// </summary>
	public class AddCourseCommand
	{
		/// <summary>
		/// Gets or sets the name.
		/// </summary>
		/// <value>
		/// The name.
		/// </value>
		public string Name { get; set; }

		/// <summary>
		/// gets or sets MaxParticipants
		/// </summary>
		public int MaxParticipants { get; set; }

		/// <summary>
		/// gets or sets Teacher
		/// </summary>
		public string Teacher { get; set; }

		/// <summary>
		/// Gets or sets the Course unique identifier.
		/// </summary>
		/// <value>
		/// The Course unique identifier.
		/// </value>
		public Guid CourseGuid { get; set; }
	}
}